### PR TITLE
stats: Add a UDS default address in addition to UDP one.

### PR DESCRIPTION
**Pull request by arguile-**
Working with UDS allows container discovery. Make it slightly easier to work with UDS and the default address used by the agent.

```go
dd := datadog.NewExporter(datadog.Options{
	StatsAddr: datadog.DefaultStatsAddrUDS,
})
```

Address taken from https://docs.datadoghq.com/developers/dogstatsd/unix_socket/.

### CODE REVIEW

File: stats.go [hunk 1]
```diff
@@ -14,8 +14,13 @@ import (
...
 )
 
-// defaultStatsAddr specifies the default address for the DogStatsD service.
-const defaultStatsAddr = "localhost:8125"
+const (
+	// DefaultStatsAddrUDP specifies the default protocol (UDP) and address for the DogStatsD service.
+	DefaultStatsAddrUDP = "localhost:8125"
+
+	// DefaultStatsAddrUDS specifies the default socket address for the DogStatsD service over UDS.
+	DefaultStatsAddrUDS = "unix:///var/run/datadog/dsd.socket"
```
Comment (gbbr) [inline stats.go]: What happens on systems that aren't UNIX? Should we move this constant into a file that builds only for supporting environments using the `+build` directive?

**Comment by arguile-** [comment]
I think it's a bit of overkill for something that should just fail during the dial if used, but it's easy enough to do.

**Comment by arguile-** [comment]
It seems like the there is some support for the unix sockets in some of the other systems.

https://golang.org/src/net/unixsock_posix.go
```go
// +build darwin dragonfly freebsd js,wasm linux nacl netbsd openbsd solaris windows
```

But is inconsistent.

https://golang.org/src/net/unixsock.go
```go
// BUG(mikio): On JS, NaCl, Plan 9 and Windows, methods and functions
// related to UnixConn and UnixListener are not implemented.
```

I can't test on any of these systems but I think the build tags I used should be reasonable as they are the common "unix" set used throughout a lot of Go.

**Comment by gbbr** [comment]
I think you're right. It _is_ a bit overkill. Sorry about that. Should we switch back? I was aiming for correctness. People viewing `godoc` will see the platform godoc is running on anyway, unless they add the `GOOS` query string which most people don't know about anyway...

**Comment by arguile-** [comment]
Reverted but added a note about platform support of unix sockets.

**Comment by gbbr** [comment]
Thanks!

**Pull request merged**